contribute/design/1.x/apis/cstor_volume.md

```diff
@@ -0,0 +1,309 @@
+	// controller(i.e resource is just created)
+	CVRStatusEmpty CStorVolumeReplicaPhase = ""
+
+	// CVRStatusOnline describes volume replica is Healthy and data existing on
+	// the healthy replica is up to date
+	CVRStatusOnline CStorVolumeReplicaPhase = "Healthy"
+
+	// CVRStatusOffline describes volume replica is created but not yet connected
+	// to the target
+	CVRStatusOffline CStorVolumeReplicaPhase = "Offline"
+
+	// CVRStatusDegraded describes volume replica is connected to the target and
+	// rebuilding from other replicas is not yet started but ready for serving
+	// IO's
+	CVRStatusDegraded CStorVolumeReplicaPhase = "Degraded"
+
+	// CVRStatusNewReplicaDegraded describes replica is recreated (due to pool
+	// recreation[underlying disk got changed]/volume replica scaleup cases) and
+	// just connected to the target. Volume replica has to start reconstructing
+	// entier data from another available healthy replica. Until volume replica
+	// becomes healthy whatever data written to it is lost(NewReplica also not part
+	// of any quorum decision)
+	CVRStatusNewReplicaDegraded CStorVolumeReplicaPhase = "NewReplicaDegraded"
+
+	// CVRStatusRebuilding describes volume replica has missing data and it
+	// started rebuilding missing data from other replicas
+	CVRStatusRebuilding CStorVolumeReplicaPhase = "Rebuilding"
+
+	// CVRStatusReconstructingNewReplica describes volume replica is recreated
+	// and it started reconstructing entier data from other healthy replica
+	CVRStatusReconstructingNewReplica CStorVolumeReplicaPhase = "ReconstructingNewReplica"
+
+	// CVRStatusError describes either volume replica is not exist in cstor pool
+	CVRStatusError CStorVolumeReplicaPhase = "Error"
+
+	// CVRStatusDeletionFailed describes volume replica deletion is failed
+	CVRStatusDeletionFailed CStorVolumeReplicaPhase = "DeletionFailed"
+
+	// CVRStatusInvalid ensures invalid resource(currently not honoring)
+	CVRStatusInvalid CStorVolumeReplicaPhase = "Invalid"
+
+	// CVRStatusInit describes CVR resource is newly created but it is not yet
+	// created zfs dataset
+	CVRStatusInit CStorVolumeReplicaPhase = "Init"
+
+	// CVRStatusRecreate describes the volume replica is recreated due to pool
+	// recreation/scaleup
+	CVRStatusRecreate CStorVolumeReplicaPhase = "Recreate"
+)
+
+// CStorVolumeReplicaStatus is for handling status of cvr.
+type CStorVolumeReplicaStatus struct {
+	Phase    CStorVolumeReplicaPhase `json:"phase"`
+	Capacity CStorVolumeCapacityAttr `json:"capacity"`
+	// LastTransitionTime refers to the time when the phase changes
+	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty"`
+	LastUpdateTime     metav1.Time `json:"lastUpdateTime,omitempty"`
+	Message            string      `json:"message,omitempty"`
+}
+
+// CStorVolumeCapacityAttr is for storing the volume capacity.
+type CStorVolumeCapacityAttr struct {
+	TotalAllocated string `json:"totalAllocated"`
```

Comment:
Should this be Total? instead of TotalAllocated? 

Also, can this be called CStorVolumeCapacityDetails

Reply:
done changed to
```go

// CStorVolumeCapacityDetails represents capacity information releated to volume
// replica
type CStorVolumeCapacityDetails struct {
	// The amount of space consumed by this volume replica and all its descendents
	Total string `json:"totalAllocated"`
	// The amount of space that is "logically" accessible by this dataset. The logical
	// space ignores the effect of the compression and copies properties, giving a
	// quantity closer to the amount of data that applications see.  However, it does
	// include space consumed by metadata
	Used string `json:"used"`
}

```